packages/localize-monetary-amount/src/localize-monetary-amount.ts

```diff
@@ -0,0 +1,980 @@
+ *       units/; e.g. in USD we have 1 dollar == 100 cents. For some currencies
+ *       the minor unit is no longer in use due to inflation or devaluation;
+ *       this is the case for JPY. In these cases (and JPY is the main one)
+ *       we might as well think of the minor unit as the same as the major unit.
+ *       Regardless of currency, monetary amounts are formatted as an integer
+ *       number of major units with any minor unit remainder as a decimal part.
+ *   * exponent
+ *       For most currencies the number of minor units per major unit is a
+ *       power of ten, either 100 = 10^2 or 1000 = 10^3. In these cases the
+ *       power (2 or 3) is called the /exponent/. HOWEVER note that there are
+ *       some currencies where the number of minor units per major unit is not
+ *       an integer power of 10, so working with the exponent (2 or 3) is harder
+ *       than working with the multiple directly (100 or 1000). This code uses
+ *       the multiple only.
+ *
+ *
+ * NOTES FOR MAINTAINERS
+ * =====================
+ *
+ * This package will almost certainly require tweaking on a locale-by-locale
+ * basis. It's intended that this will happen in two places:
+ *
+ *   * To adjust formatting for a specific locale and currency:
+ *       Look at currencyFormattingSchema(). This is a giant nested switch
+ *       statement that governs the format used for a given locale/currency pair.
+ *       See also the CurrencyFormat type.
+ *   * To add a new format:
+ *       Look at localizeCurrencyWithSchema(). This is where a format slug
+ *       and amount data get translated to an actual string.
+ *
+ *
+ * FURTHER READING
+ * ===============
+ *
+ * Surprisingly there are no standard documents on localized currency formatting,
+ * but the following resources are helpful:
+ *
+ *     https://publications.europa.eu/code/en/en-370303.htm
+ *     https://en.wikipedia.org/wiki/ISO_4217
+ *     https://en.wikipedia.org/wiki/Currency_symbol
+ *     https://en.wikipedia.org/wiki/Decimal_separator
+ *     https://www.thefinancials.com/Default.aspx?SubSectionID=curformat
+ *     https://www.loc.gov/standards/iso639-2/php/English_list.php
+ *
+ * @param {LocaleCode<Raw>} rawLocaleCode
+ *   ISO 631-1 language code with an optional ISO 3166-1 alpha-2 region code,
+ *   separated by a hyphen. Examples: 'en', 'en-gb', 'fr-be'.
+ * @param {CurrencyCode} currencyCode
+ *   ISO 4217 currency code. Examples: 'USD', 'JPY', 'BRL'.
+ * @param {number} amount
+ *   Integer amount in minor currency units. Examples: $1.00 USD and ¥100 JPY
+ *   are both passed as 100.
+ * @returns {string}
+ *   Localized monetary amount
+ */
+export function localizeMonetaryAmount(
+	rawLocaleCode: LocaleCode< Raw >,
+	currencyCode: CurrencyCode,
+	amount: number
+): LocalizedMonetaryAmount {
+	// Validate arguments
+	if ( ! ( typeof amount === 'number' && Number.isInteger( amount ) ) ) {
+		throw 'amount parameter must be an integer.';
```

Comment:
Generally [it's best](https://stackoverflow.com/questions/11502052/throwing-strings-instead-of-errors) to throw an `Error` object (or a custom error if you like); many places that catch errors expect this, and it gives helpful metadata (like a stack trace) for free.